pyiron_contrib/generic/coscineIo.py

```diff
@@ -0,0 +1,701 @@
+                            tmp_dir,
+                            "pyiron_"
+                            + self.metadata["External/alias ID"].raw()
+                            + ".h5",
+                        )
+                    )
+                    os.rename(file_name, new_name)
+                    file_name = new_name
+                self._data = file_name
+            else:
+                self._data = self._coscine_object.content()
+
+    def download(self, path="./"):
+        self._coscine_object.download(path=path)
+
+    def _get_metadata(self):
+        form = CoscineMetadata(self._coscine_object.form())
+        form.parse(self._coscine_object.metadata(force_update=True))
+        return form
+
+    def _set_metadata(self, metadata):
+        raise AttributeError("can't set attribute")
+
+
+def _list_filter(list_to_filter, **kwargs):
+    """Return entry in list matching all attributes."""
+
+    def filter_func(item):
+        for key, value in kwargs.items():
+            if getattr(item, key) != value:
+                return False
+        return True
+
+    return list(filter(filter_func, list_to_filter))
+
+
+class Job2CoscineMetadataConverter:
+
+    _known_profiles = [
+        "https://purl.org/coscine/ap/sfb1394/AtomisticSimulation/",
+        "https://purl.org/coscine/ap/sfb1394/AtomisticSimulationMD/",
+    ]
+
+    def __init__(self, job, form):
+        form_profile = form.profile.target()
+        if "sfb1394" not in form_profile:
+            raise ValueError("Unknown form, not a SFB1394 one.")
+
+        self._job = job
+        self._form = form
+
+        user = job.project_hdf5["server"]["user"]
+        now = datetime.datetime.today()
+
+        self._form["ID"] = job.name
+        self._form["External/alias ID"] = str(job.id)
+        self._form["User"] = user
+        self._form["Date"] = now
+
+        if form_profile in self._known_profiles:
+            self._form["Software IDs"] = "pyiron"
+            self._parse_db_entry()
+            self._parse_server()
+
+    def _parse_server(self):
+        self._form["Computer name"] = self._job.server._host
+        self._form["Cores"] = str(self._job.server.cores)
+
+    def _parse_db_entry(self):
+        db = self._job.database_entry
+        self._form["Status"] = db.status
+        self._form["Submission time"] = db.timestart
+        self._form["Stop time"] = db.timestop
+        self._form["Simulation type"] = db.hamilton
+        self._form["Runtime [h]"] = str(db.totalcputime / 3600.0)
+
+    @property
+    def form(self):
+        return self._form
+
+    @classmethod
+    def extract(cls, job, form):
+        self = cls(job, form)
+        return self.form
+
+
+class CoscineResource(StorageInterface):
+    def __init__(
+        self,
+        resource: Union[coscine.Resource, dict, "CoscineResource"],
+        parent_path=None,
+    ):
+        """Giving access to a CoScInE Resource to receive or upload files
+
```

Comment:
It would be nice to have minimal code examples from the notebook of how to (a) download and (b) upload data.